src/index.js

```diff
@@ -199,6 +199,7 @@ async function main(args) {
 
   if (serviceParams === undefined) {
     logAndThrow('Unable to detect service, please specify manually.')
+    process.exit(args.nonZero ? -1 : 0)
```

Comment:
If the previous line actually throws an error, this line won’t actually execute. The functionality should be in ../bin/codecov, if I’m not mistaken, however.

Reply:
No, I agree. `npm run type-check` was mad that it wasn't returning since it doesn't know that function throws. I decided I preferred extra lines that never get hit over fighting.

Reply:
That makes sense.

TypeScript is normally able to figure that out but it may have difficulty right now because everything is in JS. Written in TS, it would see a function that always throws an error and implicitly type it as returning `never`: 

```ts
function defcon(level: number): never {
  throw new Error(`DEFCON ${level}`);
}
```

Adding the corresponding `@returns` annotation to `logAndThrow`’s JSDoc comment might provide TypeScript with the information it needs to satisfy it requiring return statements:
```js
/** @returns {never} */
```

Reply:
What is the difference between void and never, in this case? Iy's normally a healthy void.

Reply:
Good question.

- A return type of `never` signifies that the function will never run to completion. Functions that always throw an error or ones that terminate the program (e.g., `process.exit`) are typed as returning `never`.
- A return type of `void` means the function does not return a (defined) value. In practice, this just means the function evaluates to `undefined`, usually because the function has no return statements or the return statements it has are empty.